### PR TITLE
Compare section with all renderers except APR

### PR DESCRIPTION
**Pull request by amaladevi-r**

### CODE REVIEW

File: src/offer/views/Config.js [hunk 1]
```diff
@@ -18,8 +18,16 @@ const layoutConfiguration = {
         "compare": {
             "header": DomainRenderers.CreditCardNameItemRenderer,
```
Comment (ksreemathi) [inline src/offer/views/Config.js]: I'm wondering if compare and row renderer are sharing the same structure.  What will happen when the view is different for compare and row for the same domain ?

File: src/offer/domain/TextAndAdditionalInfo.js [hunk 2]
```diff
@@ -10,5 +18,17 @@ export default class TextandAdditionalInfo {
...
+
+export class TextAndAdditionalInfoList extends List<TextandAdditionalInfo, string> {
+    getItemValue(domain: ?TextandAdditionalInfo): ?string {
+        if (domain == undefined) {
+            return undefined;
+        }
+        return domain.getText();
+    }
+
+    initDomain(value: any) {
```
[the remark this answers is not in the document]
Comment (amaladevi-r) [reply src/offer/domain/TextAndAdditionalInfo.js]: changed this

File: src/offer/domain/List.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        return compact(this.domains.map((domain) => {
+            return this.getItemValue(domain);
+        }));
+    }
+
+    getDomains(): Array<DT> {
+        return filter(this.domains, (domain) => this.getItemValue(domain) != undefined);
+    }
+
+    initDomain(value: any): ?DT {
```
[the remark this answers is not in the document]
Comment (amaladevi-r) [reply src/offer/domain/List.js]: value will not have the constructor reference, value is just what we get from json, so the list domain needs to know about the constructor of its items. We either need to the pass the domain type from the child or have a method like initDomain. Do let me know if you think there is some other way to do it

File: src/offer/views/Config.js (hunk 1 above)
Comment (amaladevi-r) [reply src/offer/views/Config.js]: We will have a different renderer for it, we already have this case for FirstYearFee, we use generic fee renderer in compare and FirstYearFee renderer is the row

**Pull request merged**